pavement.py

```diff
@@ -15,7 +15,9 @@
 )
 
 def run_behave_test(config, feature, task_id=0):
-    sh('CONFIG_FILE=config/%s.json TASK_ID=%s behave features/%s.feature' % (config, task_id, feature))
+    os.environ['CONFIG_FILE'] = 'config/{}.json'.format(config)
+    os.environ['TASK_ID'] = str(task_id)
```

Comment:
We are changing a global variable from threads, won't it lead to corruption?

Reply:
I have not seen any issue while testing on both OS, What kind of corruption you are talking about?

Reply:
Okay at the next command you are basically using `sh` so it will use the local copy, cool. It won't be an issue 👍

Reply:
No wait, I was correct as there is a possibility of data corruption. To better explain this look at the below snippet.

```python3
import threading, os
import time
from random import randint

def check_fn(idx):
    os.environ['TEST_IDX'] = str(idx)
    # This is your shell script
    time.sleep(randint(0, 4))
    if str(idx) != os.environ['TEST_IDX']:
        print(f"Diff: {idx}, {os.environ['TEST_IDX']}", flush=True)

th_list = []

for i in list(range(1000)):
    th = threading.Thread(target=check_fn, args=[i])
    th_list.append(th)
    th.start()

for th in th_list:
    print(f"Before {os.environ['TEST_IDX']}", flush=True)
    th.join()
    print(f"After {os.environ['TEST_IDX']}", flush=True)

print(os.environ['TEST_IDX'])
```

You will be able to see the `Diff` value because the value you set is not necessarily the value you will fetch in code.